engine/src/main/scala/cromwell/jobstore/SqlJobStore.scala

```diff
@@ -1,61 +1,79 @@
 package cromwell.jobstore
 
-import cromwell.core.{JobOutputs, WorkflowId}
+import cromwell.core.ExecutionIndex._
+import cromwell.core.WorkflowId
+import cromwell.core.simpleton.WdlValueSimpleton._
+import cromwell.core.simpleton.{WdlValueBuilder, WdlValueSimpleton}
 import cromwell.database.sql.JobStoreSqlDatabase
-import cromwell.database.sql.tables.JobStoreEntry
+import cromwell.database.sql.tables.{JobStoreEntry, JobStoreResultSimpletonEntry}
+import wdl4s.TaskOutput
+import wdl4s.types._
+import wdl4s.values.WdlPrimitive
 
 import scala.concurrent.{ExecutionContext, Future}
-import cromwell.core.ExecutionIndex._
-import spray.json._
 
 class SqlJobStore(sqlDatabase: JobStoreSqlDatabase) extends JobStore {
   override def writeToDatabase(jobCompletions: Map[JobStoreKey, JobResult], workflowCompletions: List[WorkflowId])(implicit ec: ExecutionContext): Future[Unit] = {
     for {
-      _ <- sqlDatabase.addJobStoreEntries(jobCompletions map toJobStoreEntry)
+      _ <- sqlDatabase.addJobStoreEntries(jobCompletions map toDatabase)
       _ <- sqlDatabase.removeJobResultsForWorkflows(workflowCompletions map {_.toString})
     } yield ()
   }
 
-  private def toJobStoreEntry(jobCompletion: (JobStoreKey, JobResult)): JobStoreEntry = {
-    import JobResultJsonFormatter._
+  private def toDatabase(jobCompletion: (JobStoreKey, JobResult)): (JobStoreEntry, Iterable[JobStoreResultSimpletonEntry]) = {
     jobCompletion match {
-      case (key, JobResultSuccess(returnCode, jobResult)) => JobStoreEntry(
-        key.workflowId.toString,
-        key.callFqn,
-        key.index.fromIndex,
-        key.attempt,
-        jobSuccessful = true,
-        returnCode,
-        Some(jobResult.toJson.toString),
-        None,
-        None)
-      case (key, JobResultFailure(returnCode, throwable, retryable)) => JobStoreEntry(
-        key.workflowId.toString,
-        key.callFqn,
-        key.index.fromIndex,
-        key.attempt,
-        jobSuccessful = false,
-        returnCode,
-        None,
-        Some(throwable.getMessage),
-        Some(retryable))
+      case (key, JobResultSuccess(returnCode, jobOutputs)) =>
+        val entry = JobStoreEntry(
+          key.workflowId.toString,
+          key.callFqn,
+          key.index.fromIndex,
+          key.attempt,
+          jobSuccessful = true,
+          returnCode,
+          None,
+          None)
+        val simpletons = jobOutputs.mapValues(_.wdlValue).simplify.map {
+          s => JobStoreResultSimpletonEntry(s.simpletonKey, s.simpletonValue.valueString, s.simpletonValue.wdlType.toWdlString, -1) }
```

Comment:
Uh whoops, I meant this as only a typechecking placeholder but forgot to circle back to it.  😦 

Though now that you point this out, I'm not sure what the best thing to do here is either.  I could `Option` the FK, but as the FK isn't actually optional I'm not too thrilled with that idea.  Not sure I grok your function idea, can you point to an example of that?